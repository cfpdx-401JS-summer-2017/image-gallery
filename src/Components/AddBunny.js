import React from 'react';

export default function AddBunny({ onAdd }) {
    return (
        <div>
            <h2>Add a new bunny</h2>
            <form onSubmit={event => {
                event.preventDefault();
                const form = event.target;
                const { title, description, url } = form.elements;
                onAdd(title.value, description.value, url.value);
                form.reset();
            }}>
                <label htmlFor="title">Title:</label>
                <input id="title" name="title" required /> 
                <label htmlFor="description">Description:</label>
                <input id="description" name="description" required /> 
                <label htmlFor="url">url:</label>
                <input id="url" name="url" required /> 
                <button type="submit">Add Bunny</button>
            </form>
        </div>
    );
}