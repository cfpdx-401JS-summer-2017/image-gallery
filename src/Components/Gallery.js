import React, { Component } from 'react';
import PropTypes from 'prop-types';

GalleryItem.propTypes = {
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    bunny: PropTypes.object.isRequired,
    onRemove: PropTypes.func.isRequired
}

export function GalleryItem({ title, description, url, bunny, onRemove, onUpdate, bunnyNum }) {
    return (
        <div className="galleryView">
            <img src={url} alt={description} />
            <h3>{title}</h3>
            <p>{description}</p>
            <button onClick={() => onRemove(bunny)} >X</button>
        </div>  
    );
}

export class Gallery extends Component {
    
    static propTypes = {
        galleryArray: PropTypes.arrayOf(PropTypes.object).isRequired,
        onRemove: PropTypes.func.isRequired
    }
    
    render() {
        const { galleryArray, onRemove, onUpdate, bunnyNum } = this.props;
        const bunnyGallery = galleryArray.map((bunny, i) => (
            <GalleryItem key={i} bunny={bunny} bunnyNum={bunnyNum} title={bunny.title} description={bunny.description} url={bunny.url} onRemove={onRemove} onUpdate={onUpdate} />
        ));
        return (
            <div>
                {bunnyGallery[bunnyNum]}
                <button onClick={() => onUpdate(bunnyNum - 1)}>Previous Bunny</button>
                <button onClick={() => onUpdate(bunnyNum + 1)}>Next Bunny</button>
            </div>
        );
    }
}