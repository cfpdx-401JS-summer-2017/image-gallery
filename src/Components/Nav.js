import React from 'react';
import { Link, withRouter } from 'react-router-dom';
import qs from 'qs';

function Nav({ location }) {

    const currentView = qs.parse(location.search.slice(1)).view || 'list'; 
    
    return(
        <nav id="mainNav">
            <ul>
                <li><Link to="/">Home</Link></li>
                <li><Link to="/about">About</Link></li>
                <li><Link to={`/images?view=${currentView}`}>Images</Link></li>
            </ul>
        </nav>
    );
}

export default withRouter(Nav);