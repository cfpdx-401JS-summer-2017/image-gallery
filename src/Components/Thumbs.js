import React, { Component } from 'react';
import PropTypes from 'prop-types';

ThumbItem.propTypes = {
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    bunny: PropTypes.object.isRequired,
    onRemove: PropTypes.func.isRequired
}

export function ThumbItem({ title, description, url, bunny, onRemove }) {
    return (
        <div className="thumbView">
            <a href={url} target="_blank"><img src={url} alt={description} /></a>
            <p>{description}</p>
            <button onClick={() => onRemove(bunny)} >X</button>
        </div>
    );
}

export class Thumbs extends Component {

    static propTypes = {
        thumbArray: PropTypes.arrayOf(PropTypes.object).isRequired,
        onRemove: PropTypes.func.isRequired
    }

    render() {
        const { thumbArray, onRemove } = this.props;
        return (
            <div className="clear">
                {thumbArray.map((bunny, i) => (
                    <ThumbItem key={i} bunny={bunny} title={bunny.title} description={bunny.description} url={bunny.url} onRemove={onRemove} />
                ))}
            </div>
        );
    }
}