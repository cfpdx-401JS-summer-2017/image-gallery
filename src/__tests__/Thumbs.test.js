import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { Thumbs, ThumbItem } from '../Components/Thumbs';

const testFunc = function () { }

describe('ThumbItem tests', () => {

    it('displays correct properties', () => {
        const bunny = {};
        const wrapper = shallow(<ThumbItem title="Test Title" description="Test description" url="http://google.com/" bunny={bunny} onRemove={testFunc} />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

});

describe('Thumb tests', () => {

    it('can render 3 bunnies', () => {

        const wrapper = shallow(

            <Thumbs onRemove={testFunc} thumbArray={[
                {
                    title: 'Cute Bunny 1',
                    description: 'Who\'s a cute bunny??',
                    url: 'http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg',
                    bunny: {},
                    onRemove: testFunc
                },
                {
                    title: 'Cute Bunny 2',
                    description: 'Diabolically cute!!',
                    url: 'http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-25__605.jpg',
                    bunny: {},
                    onRemove: testFunc
                },
                {
                    title: 'Cute Bunny 3',
                    description: 'There are no bounds to the cuteness!',
                    url: 'http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-110__605.jpg',
                    bunny: {},
                    onRemove: testFunc
                }
            ]} />

        )

        let thumbArray = wrapper.unrendered.props.thumbArray;
        expect(thumbArray.length).toEqual(3);

    });

});