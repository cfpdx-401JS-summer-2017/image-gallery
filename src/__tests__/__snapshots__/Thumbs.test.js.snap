// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThumbItem tests displays correct properties 1`] = `
<div
  className="thumbView"
>
  <a
    href="http://google.com/"
    target="_blank"
  >
    <img
      alt="Test description"
      src="http://google.com/"
    />
  </a>
  <p>
    Test description
  </p>
  <button
    onClick={[Function]}
  >
    X
  </button>
</div>
`;
